pkg/download/consistent_hashing.go

```diff
@@ -32,6 +32,7 @@ type ConsistentHashingMode struct {
 type CacheKey struct {
 	URL   *url.URL `hash:"string"`
 	Slice int64
+	Retry bool
```

Comment:
Maybe I'm overcomplicating things but perhaps this should be an integer `RetryCount` to allow retries to hit more than one alternative node? We can cap it at one retry elsewhere in the code if we're worried about disturbing the cache layout, but it seems that this would give us a little more flexibility in future.

Reply:
Agreed. I don't think we will go beyond 2 retries unless the cache footprint is very wide but that exposes some provider specific issues.  This keeps flexibility without needing to cause a complete cache shuffle to change the key for a retry increase.

Reply:
done. I called it `Attempt`.